docs/src/app/(public)/(content)/react/utils/use-render/demos/render-callback/css-modules/index.tsx

```diff
@@ -0,0 +1,52 @@
+'use client';
+import * as React from 'react';
+import { useRender } from '@base-ui-components/react/use-render';
+import styles from './index.module.css';
+
+type CounterState = {
+  odd: boolean;
+};
+
+type CounterProps = {
+  className?: string;
+  render?: useRender.RenderProp<CounterState>;
+};
+
+function Counter(props: CounterProps) {
+  const { render = <button type="button" />, className, ...otherProps } = props;
+  const [count, setCount] = React.useState(0);
+  const odd = count % 2 === 1;
+  const state = React.useMemo(() => ({ odd }), [odd]);
+
+  const { renderElement } = useRender({
+    render: render,
+    state,
+    props: {
+      ...otherProps,
+      className: `${styles.Button} ${className ?? ''}`,
```

Comment:
We can remove the `className`

```suggestion
  render?: useRender.RenderProp<CounterState>;
};

function Counter(props: CounterProps) {
  const { render = <button type="button" />, ...otherProps } = props;
  const [count, setCount] = React.useState(0);
  const odd = count % 2 === 1;
  const state = React.useMemo(() => ({ odd }), [odd]);

  const { renderElement } = useRender({
    render,
    state,
    props: {
      ...otherProps,
      className: styles.Button,
```

Reply:
But this way we are overriding the user's class name no? How we can omit it?

Reply:
The example doesn't pass a `className` to the Counter component anymore, so we can remove it.

Reply:
But regardless, if we are showing a usable component, people will need to do this, no?

Reply:
> people will need to do this, no?

You would also need to handle the role, keyboard interactions, etc. if you change the rendered element. But I think that goes way beyond the purpose of the demo. I consider not managing class name in the same category as this stuff. We don't manage class names in the first demo (the Text one) to keep it focused on what's important.

It also depends on wether you want to showcase the Counter component as an unstyled primitive (`styles.Button` should be passed to the component) or as a styled component (as it is currently, class name is defined inside the component). Also, not all components are meant to be overridden, it depends on the use case.

I think the `className` adds more noise to the demo (which in complete honesty, is kinda silly but I couldn't come up with something better that doesn't exist in Base UI today e.g. collapsible, toggle, etc).

Reply:
On the contrary we discussed just now that the demo should also show how people should merge props. For now I manually merged all props, but we are discussing adding this as a first class support in the hook, see https://github.com/mui/base-ui/pull/1418#issuecomment-2692010558. This way we would have feature parity with the Slot component from Radix UI.